app.rb

```diff
@@ -0,0 +1,184 @@
+    end
+    print "\n"
+  end
+
+  def create_person
+    puts 'Do you want to create'
+    puts '1.- Student'
+    puts '2.- Teacher'
+    puts '3.- Cancel'
+    puts "\n"
+
+    print 'Option: '
+    op = Integer(gets.chomp)
+
+    case op
+    when 1
+      new_student = @person_creator.create_student
+      new_student.id = generate_new_person_id(@persons)
+      print "Student created successfully\n\n"
+      new_student
+    when 2
+      new_teacher = @person_creator.create_teacher
+      new_teacher.id = generate_new_person_id(@persons)
+      print "Teacher created successfully\n\n"
+      new_teacher
+    end
+  end
+
+  def choose_book
+    option_is_valid = false
+    until option_is_valid
+      puts 'Select a book from the following list by number'
+      @books.each_with_index do |book, index|
+        print "#{index}) Title: \"#{book.title}\", Author: \"#{book.author}\"\n"
+      end
+      print 'Option: '
+      op = Integer(gets.chomp)
+      option_is_valid = true if op <= (@persons.length - 1)
+      print "Wrong item, try again \n\n" if op > (@persons.length - 1)
+    end
+    @books[op]
+  end
+
+  def choose_person
+    option_is_valid = false
+    until option_is_valid
+      puts 'Select a person from the following list by number (not id)'
+      @persons.each_with_index do |person, index|
+        print "#{index}) ID: #{person.id}, Name: #{person.name?}, Age: #{person.age?}\n"
+      end
+      print 'Option: '
+      op = Integer(gets.chomp)
+      option_is_valid = true if op <= (@books.length - 1)
+      print "Wrong item, try again\n\n" if op > (@books.length - 1)
+    end
+    @persons[op]
+  end
+
+  def create_rental
+    if @books.empty?
+      print "There are no books\n\n"
+      return
+    end
+    if @persons.empty?
+      print "There are no persons\n\n"
+      return
+    end
+    choosen_book = choose_book
+    print "\n"
+    choosen_person = choose_person
+    print "\n"
+
+    print 'Date [AAAA/MM/DD]: '
+    date = gets.chomp
+
+    print "Rental created successfully\n\n"
+    Rental.new(date, choosen_book, choosen_person)
+  end
+
+  def list_rentals
+    print 'Person ID: '
+    id = Integer(gets.chomp)
+    choosen_person = nil
+    puts 'Rentals: '
+    @persons.each do |person|
+      choosen_person = person if person.id == id
+    end
+    return if choosen_person.nil?
+
+    choosen_person.rentals.each do |rental|
+      print "Date: #{rental.date}, "
+      print "Book: #{rental.book.title}, "
+      print "Person: #{rental.person.name?}\n"
+    end
+    print "\n\n"
+  end
+
+  def run
+    op = show_menu
+
+    case op
+    when 1
+      list_books
+    when 2
+      list_persons
+    when 3
+      @persons << create_person
+    when 4
+      @books << create_book
+    when 5
+      @rentals << create_rental
+    when 6
+      list_rentals
+    else
+      puts 'bye'
+      exit
+    end
+
+    run
+  end
+end
+
+def main
+  app = App.new
+  app.run
+end
+
+main
```

Comment:
- [ ] Kindly make sure you have created a `main.rb` file. In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
- Present the user with a list of options to perform.
- Lets users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.